### DISCUSSION TITLE
Upload Speed Limited

### DISCUSSION THREAD

**Discussion by gomaaz** [ZoeyVid/NPMplus#1166] [Q&A] [answered]
I just set up a Docker container running OpenSpeedTest:

- When I access the Docker container directly via its IP on an unsecured port (bypassing Nginx Proxy Manager, NPM), I get the following speeds: 400 Mbit Upload / 200 Mbit Download.
- When the Docker container is attached to the bridged network (through NPMPlus on the host network), the speeds are cut in half: 200 Mbit Upload / 100 Mbit Download.

Key points:
- No security settings are enabled on the proxy host.
- CPU resources are sufficient.

Question:
What exactly is causing this speed limitation?
How can I control or resolve this issue?

**Comment by Zoey2936** [comment]
Can you try to disable proxy buffering via env in the compose file?

**Comment by gomaaz** [comment]
I just did: 

NGINX_DISABLE_PROXY_BUFFERING=true

Speeds are the same.


Now i notice following lines in the log:
![image](https://github.com/user-attachments/assets/c119e852-096c-4950-84fb-9d867bb2b4cb)

**Reply by gomaaz** [reply]
OK I noticed, 
those logs are also show with
NGINX_DISABLE_PROXY_BUFFERING=false

**Reply by Zoey2936** [reply]
They should not exist if it is set to true. Can you please set it to true and run inside the container: `nginx -T | grep buf`

**Reply by gomaaz** [reply]
![image](https://github.com/user-attachments/assets/77ae386d-a8e9-4fec-8ac4-570b038a9b2b)

**Reply by Zoey2936** [reply]
does the line in the log still appear?

**Reply by gomaaz** [reply]
Yes, it still shows up 

I guess it's the CPU...

As I watched the "nginx worker Process"

One of these filled up to 25%
the CPU core (1 of 4) head up to 100%

I wonder if the Worker Process is limited to an amount of cores

Resources are not limited by docker-environment.

**Reply by gomaaz** [reply]
![image](https://github.com/user-attachments/assets/bcc92cbe-fe17-4557-84a1-a0b7c41112f5)

**Reply by Zoey2936** [reply]
can you try to disable brotli and set `gzip off;` in the advanced tab?

**Comment by gomaaz** [comment]
Fullspeed with "gzip off;" 👍 

CPU is also as lowest as idle... 
Thanks!

**Reply by Zoey2936** [reply]
so I need to fix the mime types...

**Reply by Zoey2936** [reply, marked as answer]
should be fixed now and work without any advanced confg
